src/prefect/cli/auth.py

```diff
@@ -22,135 +38,265 @@ def auth():
+    \b    $ prefect auth login --key MY_KEY
 
-    \b
-        $ prefect auth switch-tenants --slug test-person
-        Tenant switched
+    \b  Log out, removing your current key or token
 
-    \b
-        $ prefect auth create-token -n MyToken -s RUNNER
-        ...token output...
+    \b    $ prefect auth logout
+
+    \b  Switch to another tenant by slug
+
+    \b    $ prefect auth switch-tenants --slug test-person
+
+    \b  Create a new API key that expires at the start of the next year
+
+    \b    $ prefect auth create-key -n marvin --expire 2022-1-1
     """
     if config.backend == "server":
-        raise click.UsageError("Auth commands with server are not currently supported.")
+        raise click.UsageError(
+            "Prefect Server does not have authentication. Change your backend to "
+            "Prefect Cloud with `prefect backend cloud` to log in."
+        )
 
 
 @auth.command(hidden=True)
 @click.option(
-    "--token", "-t", required=True, help="A Prefect Cloud API token.", hidden=True
+    "--key",
+    "-k",
+    help="A Prefect Cloud API key.",
 )
-def login(token):
+@click.option(
+    "--token",
+    "-t",
+    help="A Prefect Cloud API token. DEPRECATED.",
+)
+@handle_terminal_error
+def login(key, token):
     """
-    Log in to Prefect Cloud with an api token to use for Cloud communication.
+    Login to Prefect Cloud
 
-    \b
-    Options:
-        --token, -t         TEXT    A Prefect Cloud api token  [required]
-    """
-    check_override_auth_token()
+    Create an API key in the UI then login with it here:
 
-    client = Client(api_token=token)
+        $ prefect auth login -k YOUR-KEY
 
-    # Verify login obtained a valid api token
-    try:
-        output = client.graphql(
-            query={"query": {"user": {"default_membership": "tenant_id"}}}
-        )
+    You will be switched to the default tenant associated with the key. After login,
+    your available tenants can be seen with `prefect auth list-tenants` and you can
+    change the default tenant on this machine using `prefect auth switch-tenants`.
 
-        # Log into default membership
-        success_login = client.login_to_tenant(
-            tenant_id=output.data.user[0].default_membership.tenant_id
-        )
+    The given key will be stored on disk for later access. Prefect will default to using
+    this key for all interaction with the API but frequently overrides can be passed to
+    individual commands or functions. To remove your key from disk, see
+    `prefect auth logout`.
+
+    This command has backwards compatibility support for API tokens, which are a
+    deprecated form of authentication with Prefect Cloud
+    """
+    if not key and not token:
+        raise TerminalError("You must supply an API key or token!")
+
+    if key and token:
+        raise TerminalError("You cannot supply both an API key and token")
 
-        if not success_login:
-            raise AuthorizationError
+    abort_on_config_api_key(
```

Comment:
could we just auto remove the config?

Reply:
Naw we don't want to modify the `config.toml` programatically, we can't preserve comments easily and it's not in-line with our previous statements about it